python/cuml/test/test_train_test_split.py

```diff
@@ -87,3 +98,106 @@ def test_split_invalid_proportion(train_size):
+    for i in range(10):
+        seed_n = np.random.randint(0, int(1e9))
+        if seed_type == 'int':
+            seed = seed_n
+        if seed_type == 'cupy':
+            seed = cp.random.RandomState(seed=seed_n)
+        if seed_type == 'numpy':
+            seed = np.random.RandomState(seed=seed_n)
+        X = cudf.DataFrame({"x": range(100)})
+        y = cudf.Series(([0] * (100 // 2)) + ([1] * (100 // 2)))
+
+        X_train, X_test, y_train, y_test = train_test_split(X, y, seed=seed)
+
+        if seed_type == 'cupy':
+            seed = cp.random.RandomState(seed=seed_n)
+        if seed_type == 'numpy':
+            seed = np.random.RandomState(seed=seed_n)
+
+        X_train2, X_test2, y_train2, y_test2 = train_test_split(X, y,
+                                                                seed=seed)
+
+        assert X_train.equals(X_train2)
+        assert X_test.equals(X_test2)
+        assert y_train.equals(y_train2)
+        assert y_test.equals(y_test2)
+
+
+@pytest.mark.parametrize('type', test_array_input_types)
+@pytest.mark.parametrize('test_size', [0.2, 0.4, None])
+@pytest.mark.parametrize('train_size', [0.6, 0.8, None])
+@pytest.mark.parametrize('shuffle', [True, False])
+def test_array_split(type, test_size, train_size, shuffle):
+    X = np.zeros((100, 10)) + np.arange(100).reshape(100, 1)
+    y = np.arange(100).reshape(100, 1)
+
+    if type == 'cupy':
+        X = cp.asarray(X)
+        y = cp.asarray(y)
+
+    if type == 'numba':
+        X = cuda.to_device(X)
+        y = cuda.to_device(y)
+
+    if type == 'rmm':
+        X = rmm.to_device(X)
+        y = rmm.to_device(y)
+
+    X_train, X_test, y_train, y_test = train_test_split(X, y,
+                                                        train_size=train_size,
+                                                        test_size=test_size,
+                                                        shuffle=shuffle,
+                                                        seed=0)
+
+    if type == 'cupy':
+        assert isinstance(X_train, cp.ndarray)
+        assert isinstance(X_test, cp.ndarray)
+        assert isinstance(y_train, cp.ndarray)
+        assert isinstance(y_test, cp.ndarray)
+
+    if type in ['numba', 'rmm']:
+        assert cuda.devicearray.is_cuda_ndarray(X_train)
+        assert cuda.devicearray.is_cuda_ndarray(X_test)
+        assert cuda.devicearray.is_cuda_ndarray(y_train)
+        assert cuda.devicearray.is_cuda_ndarray(y_test)
+
+    if train_size is not None:
+        assert X_train.shape[0] == X.shape[0] * train_size
+        assert y_train.shape[0] == y.shape[0] * train_size
+
+    if test_size is not None:
+        assert X_test.shape[0] == X.shape[0] * test_size
+        assert y_test.shape[0] == y.shape[0] * test_size
+
+    if shuffle is None:
+        assert X_train == X[0:train_size]
+        assert y_train == y[0:train_size]
+        assert X_test == X[-1 * test_size:]
+        assert y_test == y[-1 * test_size:]
+
```

Comment:
As a sanity check for the shuffled case,  concatenate the train & test arrays & sort them so they can be compared for equality